scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/Readme.md

```diff
@@ -0,0 +1,172 @@
+# IPC Footprint Generator for No Lead Packages
+
+This generator uses IPC-7351B equations and fillet definitions to derive a footprint for no-lead style packages from the dimensions of the package. (The suggested footprint is only used to determine the size of the exposed pad.) Fillet size definitions can be changed by pointing to a personalized IPC parameter file. The script generates footprints in zero orientation version A (pin 1 at top left corner).
+Pads are generated using rounded rectangle pads as suggested in preliminary releases of IPC-7351C.
+
+Examples of supported packages include: DFN, QFN, SON, LGA.
+
+## Running the Script
+
+The script requires python 3.4 or newer. Run it with:
+`python3 ipc_noLead_generator.py size_definitions/dfn.yaml` (replace `dfn.yaml` with the size definition file that contains your part.)
+
+### Optional Script Parameters
+
+* --global_config: the config file defining how the footprint will look like. (KLC) (default=`../../tools/global_config_files/config_KLCv3.0.yaml`)
+* --series_config: the config file defining series parameters (footprint naming). (default=`../package_config_KLCv3.yaml`)
+
+* --density: IPC density level (L,N,M) (default=`N`)
+* --ipc_doc: IPC definition document (default=`../ipc_definitions.yaml`)
+
+* --force_rectangle_pads: Force the generation of rectangle pads instead of rounded rectangle
+* --kicad4_compatible: Create footprints compatible with version 4 (avoids round-rect and custom pads).
+
+## Size Definition Format
+
+
+The size definition file contains one entry per package size definition. The top level parameter will give the internal parameter set name. This one must be unique in this file and should be representative of the footprint as it will be used in error messages. (It also makes later maintenance easier if it is easy to determine which footprint was generated by which parameter set.)
+
+``` yaml
+internal_package_name:
+  # parameter list (See below)
+```
+
+### Documentation and Naming Parameters
+- Size source to be added to footprint documentation field (`size_source`) {url}
+- Footprint library name (`library`) {string}
+- Footprint name generation control
+  - Device type added as prefix. Example QFN. (`device_type`) {string}
+  - Selction of different name format. Example for LGA footprints. (`use_name_format`) {enum(`LGA`, `not LGA`)}
```

Comment:
First word has a typo. And perhaps replace 'different' with 'alternate'.

Reply:
Fixed